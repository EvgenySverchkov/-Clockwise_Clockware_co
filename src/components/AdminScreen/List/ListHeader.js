import React from 'react';

export default function ListHeader({templObj}){
  let keysArr = Object.keys(templObj||{});
  if(keysArr.length===0){
    return (<tr><th>List is empty</th></tr>)
  }
  return (
    <tr>
      {
        keysArr.map(item=>{
          let value = item;
          if(typeof item === 'string'){
            let upperChar = item.match(/[A-Z]/);
            let strCopy = item;
            if(upperChar){
              let idx = upperChar.index;
              strCopy = item.slice(0, idx) + " " + item[idx].toLocaleLowerCase() + item.slice(idx+1);
            }
            value = strCopy[0].toUpperCase() + strCopy.slice(1);
          }
          return (<th key={item} className="text-center" style={{wordWrap: "break-word"}}>{value}</th>);
        })
      }
      <th></th>
    </tr>
  )
}
