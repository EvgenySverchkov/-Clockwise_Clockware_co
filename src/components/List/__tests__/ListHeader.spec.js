import React from "react";
import {create} from 'react-test-renderer';

import ListHeader from "../ListHeader";

it("Test of <ListHeader />", ()=>{
    const mockProps = {
        templArr: ["name", "rating"]
    }
    const component = create(<ListHeader {...mockProps}/>);
    expect(component.toJSON()).toMatchSnapshot();
});