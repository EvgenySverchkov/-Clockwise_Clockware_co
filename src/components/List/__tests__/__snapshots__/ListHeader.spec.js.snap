// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <ListHeader /> 1`] = `
<tr>
  <th>
    Name
  </th>
  <th>
    Rating
  </th>
  <th />
</tr>
`;
